syntax = "proto3";

package dht;

service DHTService {
  rpc Put(PutRequest) returns (PutResponse);
  rpc Get(GetRequest) returns (GetResponse);
  rpc Delete(DeleteRequest) returns (DeleteResponse);
}

message Command {
  enum Type {
    PUT = 0;
    DELETE = 1;
  }
  Type type = 1;
  bytes key = 2;
  bytes value = 3;
}

message PutRequest {
  bytes key = 1;
  bytes value = 2;
}

message PutResponse {
  bool success = 1;
}

message GetRequest {
  bytes key = 1;
}

message GetResponse {
  bytes value = 1;
  bool found = 2;
}

message DeleteRequest {
  bytes key = 1;
}

message DeleteResponse {
  bool success = 1;
}